### PR TITLE
feat: validations and interactable social links

### PR DESCRIPTION
**Pull request by Sardar1208**
**- What I did - added validations for social links and email, made social links and email interactable **
ticket link - https://github.com/atsign-foundation/apps/issues/303

### CODE REVIEW

File: lib/screens/edit_persona/edit_category_fields.dart [hunk 1]
```diff
@@ -166,15 +166,13 @@ class _EditCategoryFieldsState extends State<EditCategoryFields> {
...
-                            if (customFields != null) {
-                              setState(() {
-                                customFields!.add(data);
-                                UserPreview()
-                                        .user()!
-                                        .customFields[widget.category.name] =
-                                    customFields;
-                              });
-                            }
+                            setState(() {
```
Comment (sachins-geekyants) [inline lib/screens/edit_persona/edit_category_fields.dart:169]: why did we remove customFields != null ?

**Comment by sachins-geekyants** [comment]
@Sardar1208 These validations and interactivity are only added for the mobile version ?

**Review by sonle-geekyants — DISMISSED** [review]

File: lib/desktop/screens/desktop_profile_basic_info/widgets/desktop_basic_info_widget.dart [hunk 2]
```diff
@@ -71,10 +72,18 @@ class DesktopBasicInfoWidget extends StatelessWidget {
           ),
           Expanded(
             child: Container(
-              child: Text(
-                data.value ?? '',
-                style: appTheme.textTheme.bodyText2?.copyWith(
-                  color: appTheme.primaryTextColor,
+              child: GestureDetector(
```
Comment (sonle-geekyants) [inline lib/desktop/screens/desktop_profile_basic_info/widgets/desktop_basic_info_widget.dart:84]: Could you change GestureDetector -> InkWell Widget?

Comment (Sardar1208) [reply lib/desktop/screens/desktop_profile_basic_info/widgets/desktop_basic_info_widget.dart:84]: Its just a text link, that is why I have used GestureDetecter

**Comment by sachins-geekyants** [comment]
@Sardar1208 is it ready to be  review for desktop app ?

**Comment by Sardar1208** [comment]
Yes I have added support for desktop also @sachins-geekyants

**Review by sonle-geekyants — DISMISSED** [review]

**Review by sachins-geekyants — APPROVED** [review]

**Pull request merged**